src/qibo/models/evolution.py

```diff
@@ -263,7 +263,7 @@ def _loss(params, adiabatic_evolution, h1, opt_messages, opt_history):
             log.info(f"Params: {params}  -  <H1> = {loss}")
         return loss
 
-    def minimize(self, initial_parameters, method="BFGS", options=None, messages=False):
+    def minimize(self, opt, initial_parameters, fit_options={}, messages=False):
```

Comment:
Put `None` as default value and check if it is not `None`.
https://web.archive.org/web/20200605073004/http://effbot.org/zone/default-values.htm